src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -391,14 +397,63 @@ private IAIState canGoPlanting(@NotNull final ScarecrowTileEntity currentField,
      *
      * @param position the position to check.
      * @param field    the field close to this position.
-     * @return true if should be hoed.
+     * @return position of hoeable surface or null if there is not one
      */
-    private boolean shouldHoe(@NotNull final BlockPos position, @NotNull final ScarecrowTileEntity field)
+    private BlockPos findHoeableSurface(@NotNull BlockPos position, @NotNull final ScarecrowTileEntity field)
     {
+        position = getSurfacePos(position);
+        if ((position == null)
+            || field.isNoPartOfField(world, position) 
+            || (world.getBlockState(position.above()).getBlock() instanceof CropsBlock)
+            || (world.getBlockState(position.above()).getBlock() instanceof BlockScarecrow)
+            || !world.getBlockState(position).getBlock().is(Tags.Blocks.DIRT)
+        )
+            {
```

Comment:
bracket formatting seems a bit off, if you're using IDEA, you should be able to import the formatted rules that are in the repo

Reply:
Reach me out if you want formatter for eclipse/vscode that is close to idea's in repo